backend/src/main/java/com/soen341/instagram/security/AuthorizationServerConfig.java

```diff
@@ -0,0 +1,45 @@
+package com.soen341.instagram.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerSecurityConfigurer;
+
+@Configuration
+@EnableAuthorizationServer
+public class AuthorizationServerConfig extends AuthorizationServerConfigurerAdapter
+{
+	@Autowired
+	private AuthenticationManager authenticationManager;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	@Override
+	public void configure(final AuthorizationServerSecurityConfigurer security) throws Exception
+	{
+		// returns true if user not anonymous
+		security.checkTokenAccess("isAuthenticated()");
+	}
+
+	@Override
+	public void configure(final ClientDetailsServiceConfigurer clients) throws Exception
+	{
+		clients.inMemory().withClient("instagram_client").authorizedGrantTypes("client_credentials", "password")
+				.authorities("ROLE_CLIENT", "ROLE_TRUSTED_CLIENT").scopes("read", "write", "trust")
```

Comment:
I think this would be easier to change/find if we made it a class constant

Reply:
We should avoid as much as possible to include "magic numbers" in our code. Any number passed as an argument should be a constant include at the top of the class. And all caps for the name please, with snake casing!

Reply:
I agree with Daniela. It'd be easier to keep track of our variables and easier to change them if changes are required, with the use of centralized variables. With a descriptive name, it might also make things easier to understand. The easier comprehension might not apply here, but it general it usual does.

Reply:
For the `client_id` and `client_secret` as well

Reply:
Agreed. I'll do for `client_id` , `client_secret` and token time. The other value are fixed value for spring configuration.